deps/ccommon/src/cc_print.c

```diff
@@ -62,6 +81,27 @@ cc_print_uint64(char *buf, size_t size, uint64_t n)
 }
 
 size_t
+cc_print_int64(char *buf, size_t size, int64_t n)
+{
+    size_t d;
+    uint64_t ab = abs_int64(n);
+
+    d =  digits(ab) + (n < 0);
+    if (size < d) {
+        return 0;
+    }
+
+    if (n < 0) {
+        *buf++ = '-';
+        d--;
+    }
+
+    _print_uint64(buf, d, n);
+
+    return (n < 0) ? d + 1 : d;
```

Comment:
d is the number of bytes printed, so we need to return the _original_ d not the decremented value (which is then used to print the unsigned). But it is true it can be made a little simpler. I have posted the changes to ccommon.